components/infobox/wikis/mobilelegends/infobox_hero_custom.lua

```diff
@@ -0,0 +1,180 @@
+	) then
+		table.insert(widgets, Title{name = 'Base Statistics'})
+	end
+
+	table.insert(widgets, Cell{name = 'Health', content = {_args.hp}})
+	table.insert(widgets, Cell{name = 'Health Regen', content = {_args.hpreg}})
+	table.insert(widgets, Cell{name = 'Mana', content = {_args.mana}})
+	table.insert(widgets, Cell{name = 'Mana Regen', content = {_args.manareg}})
+	table.insert(widgets, Cell{name = 'Cooldown Reduction', content = {_args.cdr}})
+	table.insert(widgets, Cell{name = 'Energy', content = {_args.energy}})
+	table.insert(widgets, Cell{name = 'Energy Regen', content = {_args.energyreg}})
+	table.insert(widgets, Cell{name = 'Attack Type', content = {_args.attacktype}})
+	table.insert(widgets, Cell{name = 'Attack Damage', content = {_args.damage}})
+	table.insert(widgets, Cell{name = 'Attack Speed', content = {_args.attackspeed}})
+	table.insert(widgets, Cell{name = 'Attack Range', content = {_args.attackrange}})
+	table.insert(widgets, Cell{name = 'Ability Power', content = {_args.ap}})
+	table.insert(widgets, Cell{name = 'Armor', content = {_args.armor}})
+	table.insert(widgets, Cell{name = 'Magic Resistance', content = {_args.magicresistance}})
+	table.insert(widgets, Cell{name = 'Movement Speed', content = {_args.movespeed}})
+	table.insert(widgets, Title{name = 'Esports Statistics'})
+
+	local stats = HeroWL.create({hero = _args.heroname or _pagename})
+	stats = mw.text.split(stats, ';')
+	local winPercentage = (tonumber(stats[1]) or 0) / ((tonumber(stats[1]) or 0) + (tonumber(stats[2]) or 1))
+	winPercentage = Math.round({winPercentage, 4}) * 100
+	local statsDisplay = (stats[1] or 0) .. 'W : ' .. (stats[2] or 0) .. 'L (' .. winPercentage .. '%)'
+	table.insert(widgets, Cell{name = 'Win Rate', content = {statsDisplay}})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'header' then
+		return {
+			Header{
+				name = _args.heroname,
+				subHeader = _args.title,
+				image = _args.image,
+				imageDefault = _args.default,
+				imageDark = _args.imagedark or _args.imagedarkmode,
+				imageDefaultDark = _args.defaultdark or _args.defaultdarkmode,
+			},
+		}
+	elseif id == 'caption' then
+		table.insert(widgets, Center{content = {_args.quote}})
+	elseif id == 'type' then
+		local breakDownContents = {}
+		local lane = _args.lane
+		if not String.isEmpty(lane) then
+			lane = '<b>Lane</b><br>' .. Template.safeExpand(_frame, 'Class icon', {lane}, '')
+			table.insert(breakDownContents, lane)
+		end
+		local primaryRole = _args.primaryrole
+		if not String.isEmpty(primaryRole) then
+			primaryRole = '<b>Primary Role</b><br>' .. Template.safeExpand(_frame, 'Class icon', {primaryRole}, '')
+			table.insert(breakDownContents, primaryRole)
+		end
+		local secondaryRole = _args.secondaryrole
+		if not String.isEmpty(secondaryRole) then
+			secondaryRole = '<b>Secondary Role</b><br>' .. Template.safeExpand(_frame, 'Class icon', {secondaryRole}, '')
+			table.insert(breakDownContents, secondaryRole)
+		end
```

Comment:
Can we change Template:Class icon into a module instead?

Reply:
yes please